logging/src/lib.rs

```diff
@@ -0,0 +1,340 @@
+//! Simple timely logging
+
+extern crate time;
+extern crate abomonation;
+#[macro_use] extern crate abomonation_derive;
+
+use std::rc::Rc;
+use std::cell::RefCell;
+
+static mut PRECISE_TIME_NS_DELTA: Option<i64> = None;
+
+/// Returns the value of an high resolution performance counter, in nanoseconds, rebased to be
+/// roughly comparable to an unix timestamp.
+/// Useful for comparing and merging logs from different machines (precision is limited by the
+/// precision of the wall clock base; clock skew effects should be taken into consideration).
+#[inline(always)]
+pub fn get_precise_time_ns() -> u64 {
+    let delta = unsafe {
+        *PRECISE_TIME_NS_DELTA.get_or_insert_with(|| {
+            let wall_time = time::get_time();
+            let wall_time_ns = wall_time.nsec as i64 + wall_time.sec * 1000000000;
+            time::precise_time_ns() as i64 - wall_time_ns
+        })
+    };
+    (time::precise_time_ns() as i64 - delta) as u64
+}
```

Comment:
Might this be a good time to sort out what notion of time we want to use in these logging channels? Way back when, I floated an alternative that we use a "timely-local" time, roughly "nanoseconds since initial synchronization" rather than "local system clock", which has some good things and some bad things.

1. Good: it meant that clock skew and such were less problematic (if two computers clocks are off by a minute, this approach will stall a bunch of logs, right?).

2. Bad: it is harder to correlate events in the logging stream with other events that occur on the system, outside of timely (e.g. spikes measured through other tools).

I think the bad thing could in principle be fixed by capturing the machine local time at which the synchronization happens and disseminating this to everyone (playing the role of the `PRECISE_TIME_NS_DELTA` static). It would mean we need to write down the captured local time, to support correlation with other events that only have local time, but that seems doable.

Also we can ditch the `time` crate, which is only included in timely for this, I think.